runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -761,36 +751,32 @@ public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V>
      */
     @SuppressWarnings("unchecked")
     protected void unmapSecondaryIndexes(K key, V value) {
+        if (value == null) {
+            log.trace("Attempting to build an index with a null value, skipping.");
+            return;
+        }
+
         try {
-            if (value != null) {
-                // Map entry into secondary indexes
-                for (Index<K, V, ? extends Comparable> index : indexSpec) {
-                    String indexName = index.getName().get();
-                    Map<Comparable, Map<K, V>> secondaryIndex = secondaryIndexes.get(indexName);
-                    if (index.isMonoIndex()) {
-                        Comparable indexKey = index.indexFunction.apply(key, value);
-                        Map<K, V> slot = secondaryIndex.get(indexKey);
-                        if (slot != null) {
-                            slot.remove(key, value);
-                        }
-                    } else {
-                        for (Comparable<?> indexKey
-                                : index.getMultiValueIndexFunction().apply(key, value)) {
-                            Map<K, V> slot = secondaryIndex.get(indexKey);
-                            if (slot != null) {
-                                slot.remove(key, value);
-                            }
-                        }
+            // Map entry into secondary indexes
+            for (Index<K, V, ? extends Comparable> index : indexSpec) {
+                String indexName = index.getName().get();
+                Map<Comparable, Map<K, V>> secondaryIndex = secondaryIndexes.get(indexName);
+                for (Comparable indexKey : index.getMultiValueIndexFunction().apply(key, value)) {
+                    Map<K, V> slot = secondaryIndex.get(indexKey);
+                    if (slot != null) {
+                        slot.remove(key, value);
                     }
                 }
             }
         } catch (Exception e) {
-            indexSpec.clear();
-            secondaryIndexes.clear();
-            indexGenerationFailed = true;
-            log.error("unmapSecondaryIndexes: Exception unmapping {}, {},"
-                            + " UNMAPPING ALL INDEXES, indexing is disabled",
-                    key, value, e);
+            log.error("Received an exception while computing the index. " +
+                    "This is most likely an issue with the client's indexing function. {}", e);
```

Comment:
Please delete `{}`
![image](https://user-images.githubusercontent.com/314292/60566253-2dd32180-9d1b-11e9-89f4-4ff7b7db591d.png)

Reply:
You can install `sonar` plugin into IDEA, it will highlight all that kind of issues

Reply:
Oh, cool.